### PR TITLE
add authorization doc

### LINKED ISSUE #11: Add recommendations for authorization
Create new documentation in the solutions folder for authorization aspects/patterns.

Orientations:
https://github.com/devonfw/devon4j/blob/master/documentation/guide-access-control.asciidoc

### PR DESCRIPTION
**Pull request by lilliCao**
Resolves #11

### CODE REVIEW

File: solutions/authorization/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+* We define groups that combine permissions to typical and useful sets for the users.
+* We define roles as specific groups as required by our business demands.
+* We allow to associate users with a list of Access Controls.
+* For authorization of an implemented use case we determine the required permission. Furthermore, we determine the current user and verify that the required permission is contained in the tree spanned by all his associated Access Controls. If the user does not have the permission we throw a security exception and thus abort the operation and transaction.
+* We avoid negative permissions, that is a user has no permission by default and only those granted to him explicitly give him additional permission for specific things. Permissions granted can not be reduced by other permissions.
+* Technically we consider permissions as a secret of the application. Administrators shall not fiddle with individual permissions but grant them via groups. So the access management provides a list of strings identifying the Access Controls of a user. The individual application itself contains these Access Controls in a structured way, whereas each group forms a permission tree.
+
```
Comment (SchettlerKoehler) [inline solutions/authorization/index.asciidoc:47]: Do not use the pattern that no configured permissions mean the user has all permissions

**Review by SchettlerKoehler — CHANGES_REQUESTED** [review]

File: solutions/authorization/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+**Clarification of terms:**
+
+For clarification we also want to give a common understanding of related terms that have no unique definition and consistent usage in the wild.
+
+.Security terms related to authorization
+[options="header", cols="15%,85%"]
+|=======================
+|*Term*|*Meaning and comment*
+|Permission|A permission is an object that allows a principal to perform an operation in the system. This permission can be _granted_ (give) or _revoked_ (taken away). Sometimes people also use the term _right_ what is actually wrong as a right (such as the right to be free) can not be revoked.
```
[the remark this answers is not in the document]
Comment (lilliCao) [reply solutions/authorization/index.asciidoc:28]: |Pricipal| A user or application that can authenticate itself?

File: solutions/authorization/index.asciidoc [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+**Definition:**
+
+> Authorization is the verification that an authenticated user is allowed to perform the operation he intends to invoke.
+
+**Clarification of terms:**
+
+For clarification we also want to give a common understanding of related terms that have no unique definition and consistent usage in the wild.
+
+.Security terms related to authorization
+[options="header", cols="15%,85%"]
```
[the remark this answers is not in the document]
Comment (lilliCao) [reply solutions/authorization/index.asciidoc:25]: I think the table is fine

File: solutions/authorization/index.asciidoc (hunk 2 above)
[the remark this answers is not in the document]
Comment (SchettlerKoehler) [reply solutions/authorization/index.asciidoc:30]: https://en.wikipedia.org/wiki/Principal_(computer_security)

**Review by SchettlerKoehler — APPROVED** [review]

**Pull request merged**